control_protocol.md

```diff
@@ -0,0 +1,478 @@
+# Open Screen Control Protocol
+
+This document describes a control protocol suitable to support the
+implementation of
+the [Presentation API](https://w3c.github.io/presentation-api/) and
+the [Remote Playback API](https://w3c.github.io/remote-playback/).  It is
+intended to be implemented on top of one of the proposed Open Screen transport
+mechanisms such as [QUIC](quic.md) or [RTCDataChannel](datachannel.md).
+
+The control protocol is responsible for mapping Web application requests and
+data provided by Web applications onto a network transport for transmission
+between controlling user agents (controllers) and receiving user agents
+(presentations screens and remote playback devices).
+
+This document focuses on the syntax and semantics of the network messages
+themselves.  A future update will map these messages onto specific steps in the
+algorithms described in the respective specifications for the Presentation API
+and the Remote Playback API.
+
+## Requirements
+
+* The control protocol must implement
+  the [functional requirements for the Presentation API](requirements.md#presentation-api-requirements).
+* The control protocol must implement
+  the [functional requirements for the Remote Playback API](requirements.md#remote-playback-api-requirements).
+* The control protocol must meet [non-functional requirements](requirements.md#non-functional-requirements).
+
+If an [RTCDataChannel](datachannel.md) is used as the transport, additional
+messages will be required to exchange WebRTC signaling messages to establish the
+channel.
+
+## Message Transport
+
+To meet the requirements above, a message oriented control protocol is
+necessary.  That means that each party should be able to transmit a sequence of
+variable-length messages from the other, and have every message received by the
+other party, intact and in-order.
+
+The RTCDataChannel does support variable length messages (**TODO:** find a spec
+for it!), although there are
```

Comment:
If I understand correctly, [draft-ietf-rtcweb-data-channel-13](https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13) defines the message limits of data channels. [Section 4](https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13#section-4) defines the requirements:

> Req. 8:   The data channel transport protocol should support unbounded-length "messages" (i.e., a virtual socket stream) at the application layer, for such things as image-file-transfer; Implementations might enforce a reasonable message size limit.

On the other hand, [Section 6.6](https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13#section-6.6) illustrates message interleaving to avoid monopolization of data channel:

> The SCTP base protocol specified in [RFC4960] does not support the interleaving of user messages.  Therefore sending a large user message can monopolize the SCTP association.  To overcome this limitation, [I-D.ietf-tsvwg-sctp-ndata] defines an extension to support message interleaving, which SHOULD be used.  As long as message interleaving is not supported, the sender SHOULD limit the maximum message size to 16 KB to avoid monopolization.

Reply:
Thanks.  I think I was looking for [SCTP](https://tools.ietf.org/html/rfc4960) as the layer that provides message framing.